File: backend/app/api/customers.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session

from app.database.connection import get_db
from app.schemas.customer import CustomerCreate, CustomerResponse
from app.services.customer_service import CustomerService


router = APIRouter(
    prefix="/customers",
    tags=["Customers"],
)


@router.post("/", response_model=CustomerResponse)
def create_customer(
    data: CustomerCreate,
    db: Session = Depends(get_db),
):

    return CustomerService.create(db, data)


@router.get("/{customer_id}")
def get_customer(
    customer_id: int,
    db: Session = Depends(get_db),
):

    customer = CustomerService.get(db, customer_id)

    if not customer:
        raise HTTPException(
            status_code=404,
            detail="Customer not found",
        )

    return customer

File: frontend/src/api/customers.js
import client from "./client";

/**
 * POST /customers/
 * body: { name, email, income, credit_score }
 * returns: CustomerResponse { id, name, email, income, credit_score }
 */
export function createCustomer(data) {
  return client.post("/customers/", data).then((res) => res.data);
}

/**
 * GET /customers/{customer_id}
 */
export function getCustomer(customerId) {
  return client.get(`/customers/${customerId}`).then((res) => res.data);
}
